instrumentation/events/lttng-module/net.h

```diff
@@ -148,6 +312,31 @@ LTTNG_TRACEPOINT_EVENT_CLASS(net_dev_template,
 		ctf_integer_hex(void *, skbaddr, skb)
 		ctf_integer(unsigned int, len, skb->len)
 		ctf_string(name, skb->dev->name)
+		ctf_enum(net_network_header, unsigned char, network_header_type, __get_network_header_type(skb))
```

Comment:
Yes, the clean way the kernel does is that each layer has the information about the next layer's protocol. And at each layer, there's just so many possible protocols. As for supporting physical layer, I can't reproduce with current kernel, but back when I first played with this (kernel 3.don't remember), when the net_dev_queue tracepoint was triggered the physical layer data was not available yet, it came later in the code.

To know what we support and have an enum for those supported protocols is the easiest way to do this.